applications/asset_tracker_v2/doc/cloud_module.rst

```diff
@@ -0,0 +1,209 @@
+
+If at any point the module is disconnected, it will try to reconnect given that an LTE connection is still valid.
+The number of reconnection attempts can be adjusted by configuring the :ref:`CONFIG_CLOUD_CONNECT_RETRIES <CONFIG_CLOUD_CONNECT_RETRIES>` option.
+Reconnection is implemented with a binary backoff based on the following lookup table:
+
+.. code-block:: c
+
+   static struct cloud_backoff_delay_lookup backoff_delay[] = {
+      { 32 }, { 64 }, { 128 }, { 256 }, { 512 },
+      { 2048 }, { 4096 }, { 8192 }, { 16384 }, { 32768 },
+      { 65536 }, { 131072 }, { 262144 }, { 524288 }, { 1048576 }
+   };
+
+If the module reaches the maximum number of reconnection attempts, an error event of type :c:enum:`CLOUD_EVT_ERROR` is notified to the rest of the application, causing the application to perform a reboot.
+
+Configuration options
+*********************
+
+.. _CONFIG_CLOUD_THREAD_STACK_SIZE:
+
+CONFIG_CLOUD_THREAD_STACK_SIZE - Cloud module thread stack size
+   This option increases the cloud module's internal thread stack size.
+
+.. _CONFIG_CLOUD_CLIENT_ID_USE_CUSTOM:
+
+CONFIG_CLOUD_CLIENT_ID_USE_CUSTOM - Configuration for enabling the use of a custom cloud client ID
+   This option is used to enable the use of a custom client ID for connection to the respective cloud service.
+   By default, the cloud module uses the IMEI of the nRF9160-based device as the client ID.
+
+.. _CONFIG_CLOUD_CLIENT_ID:
+
+CONFIG_CLOUD_CLIENT_ID - Configuration for providing a custom cloud client ID
+   This option sets the custom client ID for the respective cloud service.
+   For setting a custom client ID, you need to set :kconfig:`CONFIG_CLOUD_CLIENT_ID_USE_CUSTOM` to ``y``.
+
+.. _CONFIG_CLOUD_CONNECT_RETRIES:
+
+CONFIG_CLOUD_CONNECT_RETRIES - Configuration that sets the number of cloud reconnection attempts
+   This option sets the number of times that a connection will be re-attempted upon a disconnect from the cloud service.
+
+.. _mandatory_config:
+
+Mandatory configurations
+========================
+
+To be able to use a supported cloud client library, a set of mandatory Kconfig options needs to be set.
+These typically include the cloud service hostname and the security tag associated with the certificates used to establish a connection.
+The aforementioned certificates needs to provisioned to the modem using the same security tag, prior to running the application.
+For more information on how to setup a connection and provision certificates to the modem, see the documentation for the respective client library listed in :ref:`Integration layers <integration_layers>`.
+
+.. note::
+   The :ref:`lib_nrf_cloud` library does not require any mandatory configurations to be set.
+   The nRF9160 on the DK and Thingy91 comes preprovisioned with certificates required to establish a connection to nRF Cloud.
+   The default configuration of the :ref:`lib_nrf_cloud` library uses the security tag that the nRF Cloud certificates are stored to.
```

Comment:
There is something odd here, but I cannot tell what it is. Are the nRF Cloud certificates stored to a security tag? Or are they stored in the device using the security tag? I just don't get the idea here.

Reply:
You could say that nRF Cloud certificates are stored to a security tag